src/lib/registration/registration_ui.rb

```diff
@@ -294,13 +294,16 @@ def register_selected_addons(selected_addons, known_reg_codes)
       registration_order = selected_addons.clone
 
       product_succeed = registration_order.map do |product|
-        ConnectHelpers.catch_registration_errors(
+        registered = ConnectHelpers.catch_registration_errors(
           message_prefix: "#{product.label}\n") do
-          register_selected_addon(product, known_reg_codes[product.identifier])
-        end
+            register_selected_addon(product, known_reg_codes[product.identifier])
+          end
 
         # remove from selected after successful registration
-        selected_addons.reject! { |selected| selected.identifier == product.identifier }
+        if registered
+          selected_addons.reject! { |selected| selected.identifier == product.identifier }
+        end
+        registered
```

Comment:
Previously it returned `true` when the product was successfully removed from the selected list. That should happen in all cases, but it ignored the addon registration result. Now it honors the registration result which looks as a good change to me.